internal/contour/metrics_test.go

```diff
@@ -413,9 +413,7 @@ func TestHTTPProxyMetrics(t *testing.T) {
 			},
 			Valid:    map[metrics.Meta]int{},
 			Orphaned: map[metrics.Meta]int{},
-			Root: map[metrics.Meta]int{
-				{Namespace: "finance"}: 1,
-			},
+			Root:     map[metrics.Meta]int{},
```

Comment:
It's because the object itself wasn't available; I had to change the order of things at the top of `computeRoutes` in `httpproxy_processor.go` to enable the vhost info to propagate out correctly. 

The remaining changes in `metrics_test.go` are because we actually emit more info now about the vhost. In addition, there was one time when we were actually counting things incorrectly.